kratos/includes/mesh.h

```diff
@@ -468,6 +480,13 @@ class Mesh : public DataValueContainer, public Flags
         return *i;
     }
 
+    const ElementType& GetElement(IndexType ElementId) const
+    {
+        auto i = mpElements->find(ElementId);
+        KRATOS_ERROR_IF(i == mpElements->end()) << " element index not found: " << ElementId << ".";
```

Comment:
i would say that if you do

     return mpElements[ElementId] 

it would only compile if ElementId already exists, since otherwise it would need modifying mpElements and hence the mesh. I can be wrong though...

Reply:
I checked and it does compile. I think because the compiler only checks that the pointer is not modified (not the object pointed to). But then the function behavior changes from the original since it will insert missing elements.